src/Location/WorkingSchedule.php

```diff
@@ -317,27 +325,44 @@ public function checkStatus($dateTime = null)
      */
     public function getTimeslot(int $interval = 15, DateTime $dateTime = null, int $leadTimeMinutes = 25)
     {
-        $dateTime = Carbon::instance($this->parseDate($dateTime));
+        $selectedDateTime = Carbon::instance($this->parseDate($dateTime));
+        $dateTime = Carbon::instance($this->parseDate($dateTime))->subDay();
         $interval = new DateInterval('PT'.($interval ?: 15).'M');
 
-        $timeslot = [];
-        $datePeriod = $this->createPeriodForTimeslot($dateTime, $interval);
+        $timeslots = [];
+        $datePeriod = $this->createPeriodForDays($dateTime);
+
+        $lastIntervalTime = $this->nextCloseAt($selectedDateTime->addDays($this->days));
+        if ($lastIntervalTime->lt($selectedDateTime))
+            $lastIntervalTime->addDay();
+
         foreach ($datePeriod as $date) {
-            $workingTime = WorkingTime::create($date->format('H:i'));
+
             $workingPeriod = $this->forDate($date);
 
-            if (!$workingPeriod->isOpenAt($workingTime))
-                continue;
+            foreach ($workingPeriod as $workingRange) {
 
-            if (!$this->isTimeslotValid($date, $dateTime, $leadTimeMinutes))
-                continue;
+                $startTime = $date->copy()->setTimeFromTimeString($workingRange->start()->format('H:i:s'));
+                $endTime = $date->copy()->setTimeFromTimeString($workingRange->end()->format('H:i:s'));
+                if ($workingRange->endsNextDay())
+                    $endTime->addDay(1);
 
-            $dateString = $date->toDateString();
+                $timePeriod = new DatePeriod($startTime, $interval, $endTime);
+                foreach ($timePeriod as $timeslot) {
 
-            $timeslot[$dateString][$date->getTimestamp()] = $date;
+                    if ($lastIntervalTime->lt($timeslot) OR !$this->isTimeslotValid($timeslot, $dateTime, $leadTimeMinutes))
+                        continue;
+
+                    $dateString = $date->toDateString();
+
+                    $timeslots[$dateString][$timeslot->getTimestamp()] = $timeslot;
+
+                }
+
+            }
         }
 
-        return collect($timeslot);
+        return collect($timeslots);
```

Comment:
Trying to wrap my head around this, can you please explain the changes you've made here, what the issue was and what was resolved.. I know there's a related issue but its a long thread.